### PR TITLE
Detect if removed page was visible

### LINKED ISSUE #289: Show some message when no tab is focused
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

When removing the drive that you are looking at, you get left with a blank screen (sidebar is still there)

### Describe the solution you'd like

Show a message like "drive was removed, please select smthn to look at"

### Describe alternatives you've considered

Auto select a new drive to open? could be confusing if the drive is only temporarily dc'd

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jojo2357**
Fixes #289 

I took the liberty of encapsulating the visible-page-was-removed logic into a single function and upgraded the removal process for networks and batteries.

### CODE REVIEW

**Comment by nokyan** [comment]
Hi, thank you for the PR!
This looks good so far but one small issue: when a drive was ejected, which caused the odd behaviour described in #289, the view itself is changed to Apps but the sidebar's selection doesn't reflect that change and nothing is selected instead.

File: src/ui/window.rs [hunk 1]
```diff
@@ -584,6 +584,18 @@ impl MainWindow {
...
 
+    /// Wrapper to remove page, and check if removed page was visible with global default behavior
+    fn remove_page(&self, page: ToolbarView) {
+        let imp = self.imp();
+        imp.content_stack.remove(&page);
+
+        // no visible child exists
+        if imp.content_stack.is_child_visible() {
+            imp.content_stack
+                .set_visible_child(&imp.applications_page.child());
```
Comment (nokyan) [inline src/ui/window.rs:595]: Add `imp.resources_sidebar.set_selected_list_item_by_tab_id(applications::TAB_ID);` after this line to also change the sidebar's selection.

Comment (jojo2357) [reply src/ui/window.rs]: Or just replace it instead of two calls. Seems to do the trick

**Comment by nokyan** [comment]
Thank you!

**Comment by jojo2357** [comment]
Always a pleasure.

**Pull request merged**